### PR TITLE
Delete substitution and fix delete setters

### PR DESCRIPTION
**Pull request by phanimarupaka**
@mortent @jijiew 

This PR implements the delete-subst feature. Currently delete-setter command deletes only the setter references of the fields whose filedName is same as setter name. This PR also fixes issues with delete-setter command.

### CODE REVIEW

File: kyaml/yaml/rnode.go [hunk 1]
```diff
@@ -384,6 +384,19 @@ func (rn *RNode) Fields() ([]string, error) {
 	return fields, nil
 }
 
+// FieldNodes returns the list of field nodes for a MappingNode.
+// Returns an error for non-MappingNodes.
+func (rn *RNode) FieldNodes() ([]*yaml.Node, error) {
```
Comment (monopole) [inline kyaml/yaml/rnode.go:389]: regression test coverage please

Comment (phanimarupaka) [reply kyaml/yaml/rnode.go]: Done. Thank you

File: cmd/config/internal/commands/cmddeletesubstitution.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	"sigs.k8s.io/kustomize/kyaml/openapi"
+	"sigs.k8s.io/kustomize/kyaml/setters2/settersutil"
+)
+
+// NewDeleteRunner returns a command runner.
+func NewDeleteSubstitutionRunner(parent string) *DeleteSubstitutionRunner {
+	r := &DeleteSubstitutionRunner{}
+	c := &cobra.Command{
+		Use:     "delete-subst DIR NAME",
+		Args:    cobra.MinimumNArgs(2),
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply cmd/config/internal/commands/cmddeletesubstitution.go]: Nope. Corrected.

File: kyaml/setters2/delete.go [hunk 3]
```diff
@@ -38,48 +32,62 @@ func (d *Delete) visitSequence(_ *yaml.RNode, _ string, _ *openapi.ResourceSchem
...
+func (d *Delete) visitMapping(object *yaml.RNode, _ string, _ *openapi.ResourceSchema) error {
+	fieldNodes, err := object.FieldNodes()
+	if err != nil {
+		return err
+	}
+
+	// for each of the field node key visit it as scalar to delete the array setter comment
+	for _, fieldNode := range fieldNodes {
+		fieldRNode := &yaml.RNode{}
+		fieldRNode.SetYNode(fieldNode)
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply kyaml/setters2/delete.go]: Done. Moved the logic to fieldRnodes helper method.

File: cmd/config/internal/commands/cmddeletesubstitution_test.go [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+		args              []string
+		schema            string
+		out               string
+		inputOpenAPI      string
+		expectedOpenAPI   string
+		expectedResources string
+		err               string
+	}{
+		{
+			name: "delete only subst if setter has same name - long ref",
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply cmd/config/internal/commands/cmddeletesubstitution_test.go:32]: Done. Modified one of the tests to have the . in subst.

**Pull request merged**